packages/engine-core/src/binary/BinaryEngine.ts

```diff
@@ -931,12 +932,19 @@ You very likely have the wrong "binaryTarget" defined in the schema.prisma file.
     return this.lastVersion
   }
 
-  async request<T>(
-    query: string,
-    headers: QueryEngineRequestHeaders = {},
-    _transaction?: InteractiveTransactionOptions<undefined>,
+  async request<T>({
+    query,
+    headers = {},
     numTry = 1,
-  ): Promise<QueryEngineResult<T>> {
+    clientMethod,
+    _transaction,
+  }: {
+    query: string
+    headers: QueryEngineRequestHeaders
+    numTry: number
+    clientMethod: string
+    _transaction?: InteractiveTransactionOptions<undefined>
```

Comment:
```suggestion
    transaction?: InteractiveTransactionOptions<undefined>
```

or we could even drop it here